jib-core/src/main/java/com/google/cloud/tools/jib/configuration/DockerHealthCheck.java

```diff
@@ -0,0 +1,184 @@
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.configuration;
+
+import com.google.common.collect.ImmutableList;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+/** Configuration information for performing healthchecks on a Docker container. */
+public class DockerHealthCheck {
+
+  /** Builds the immutable {@link DockerHealthCheck}. */
+  public static class Builder {
+
+    private final ImmutableList<String> command;
+    @Nullable private Duration interval;
+    @Nullable private Duration timeout;
+    @Nullable private Duration startPeriod;
+    @Nullable private Integer retries;
+
+    private Builder(ImmutableList<String> command) {
+      this.command = command;
+    }
+
+    /**
+     * Sets the time between healthchecks.
+     *
+     * @param interval the duration to wait between healthchecks.
+     * @return this
+     */
+    public Builder setInterval(Duration interval) {
+      this.interval = interval;
+      return this;
+    }
+
+    /**
+     * Sets the time until a healthcheck is considered hung.
+     *
+     * @param timeout the duration to wait until considering the healthcheck to be hung.
+     * @return this
+     */
+    public Builder setTimeout(Duration timeout) {
+      this.timeout = timeout;
+      return this;
+    }
+
+    /**
+     * Sets the initialization time to wait before using healthchecks.
+     *
+     * @param startPeriod the duration to wait before using healthchecks
+     * @return this
+     */
+    public Builder setStartPeriod(Duration startPeriod) {
+      this.startPeriod = startPeriod;
+      return this;
+    }
+
+    /**
+     * Sets the number of times to retry the healthcheck before the container is considered to be
+     * unhealthy.
+     *
+     * @param retries the number of retries before the container is considered to be unhealthy
+     * @return this
+     */
+    public Builder setRetries(int retries) {
+      this.retries = retries;
+      return this;
+    }
+
+    public DockerHealthCheck build() {
+      return new DockerHealthCheck(command, interval, timeout, startPeriod, retries);
+    }
+  }
+
+  /**
+   * Creates a disabled {@link DockerHealthCheck}.
+   *
+   * @return the new {@link DockerHealthCheck}
+   */
+  public static DockerHealthCheck disabled() {
+    return new DockerHealthCheck(ImmutableList.of("NONE"), null, null, null, null);
+  }
+
+  /**
+   * Creates a new {@link DockerHealthCheck.Builder} with the command set to be inherited from the
```

Comment:
I don't think we should use `withXYZ` to create new builder instances. But I can't think of an alternative yet.

Reply:
maybe `as`?

Reply:
Maybe if we put "builder" in the name? e.g. `builderWithInheritedCommand()`, but maybe that's too verbose. Or `inheritedCommandBuilder()`.

Reply:
I think starting with a common words makes finding them a little easier since it groups them together? So maybe `builderWith*` is okay.